samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,274 @@
+
+                JObject authentication = null;
+
+                if (value["authentication"] != null)
+                {
+                    authentication = JsonConvert.DeserializeObject<JObject>(value["authentication"].ToString());
+                }
+
+                // When adaptiveCard/action invoke activity from teams contains 6 digit state in response to nominal sign in flow from bot.
+                string state = null;
+                if (value["state"] != null)
+                {
+                    state = value["state"].ToString();
+                }
+
+                // authToken and state are absent, handle verb
+                if (authentication == null && state == null)
+                {
+                    switch (verb)
+                    {   //when token is absent in the invoke. We can initiate SSO in response to the invoke
+                        case "initiateSSO":
+                            return await initiateSSOAsync(turnContext, cancellationToken);
+                    }
+                }
+                else
+                {
+                    return createAdaptiveCardInvokeResponseAsync(authentication, state);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Authentication success.
+        /// AuthToken or state is present. Verify token/state in invoke payload and return AC response
+        /// </summary>
+        /// <param name="authentication">authToken are absent, handle verb</param>
+        /// <param name="state">state are absent, handle verb</param>
+        /// <param name="turnContext">The context for the current turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <param name="isBasicRefresh">Refresh type</param>
+        /// <param name="fileName">adaptiveCardResponseJson.json</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private InvokeResponse createAdaptiveCardInvokeResponseAsync(JObject authentication, string state, bool isBasicRefresh = false, string fileName = "adaptiveCardResponseJson.json")
+        {
+            // Verify token is present or not.
+
+            bool isTokenPresent = authentication != null ? true : false;
+            bool isStatePresent = state != null && state != "" ? true : false;
+
+            string[] filepath = { ".", "Resources", fileName };
+
+            var adaptiveCardJson = File.ReadAllText(Path.Combine(filepath));
+            AdaptiveCardTemplate template = new AdaptiveCardTemplate(adaptiveCardJson);
+            var authResultData = isTokenPresent ? "SSO success" : isStatePresent ? "OAuth success" : "SSO/OAuth failed";
+            
+            if (isBasicRefresh)
+            {
+                authResultData = "Refresh done";
+            }
+            
+            var payloadData = new
```

Comment:
adaptiveCardResponsejson.json does not have authResult data. Add this member to the json file

Like this,

{
      "type": "TextBlock",
      "text": "${authResult}",
      "size": "Medium",
      "weight": "Bolder"
    }

Reply:
We removed the authResult value (after a refresh) and replaced it with You are successfully logged in!